standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreTServerSocket.java

```diff
@@ -29,8 +32,32 @@
  * accepted sockets.
  *
  */
-public class TServerSocketKeepAlive extends TServerSocket {
-  public TServerSocketKeepAlive(TServerSocket serverSocket) throws TTransportException {
+public class MetastoreTServerSocket extends TServerSocket {
+  private final Configuration configuration;
+
+  public MetastoreTServerSocket(TServerSocket serverSocket,
+      Configuration conf) throws TTransportException {
     super(serverSocket.getServerSocket());
+    this.configuration = Objects.requireNonNull(conf);
+  }
+
+  @Override
+  public TSocket accept() throws TTransportException {
+    TSocket ts = super.accept();
+    boolean keepAlive = MetastoreConf.getBoolVar(configuration, MetastoreConf.ConfVars.TCP_KEEP_ALIVE);
+    if (keepAlive) {
+      try {
+        ts.getSocket().setKeepAlive(true);
+      } catch (SocketException e) {
+        throw new TTransportException(e);
+      }
+    }
+    // for testing purpose, get the maxThriftMessageSize from the configuration for every new connection
```

Comment:
we shouldn't use suboptimal design just for the sake of tests

Reply:
For the sake of tests, I make the `maxThriftMessageSize` can change dynamically based on the test need, compared to initialize the value at HMS startup and add an extra method to change this limit for test only.